Content.Server/CartridgeLoader/Cartridges/MessagesCartridgeSystem.cs

```diff
@@ -0,0 +1,271 @@
+using Content.Shared.CartridgeLoader;
+using Content.Shared.CartridgeLoader.Cartridges;
+using Content.Shared.PDA;
+using Robust.Shared.Containers;
+using Robust.Shared.Map;
+using Content.Server.Radio.Components;
+using Content.Server.GameTicking;
+using Robust.Shared.Timing;
+using Content.Shared.Access.Components;
+using Content.Server.Radio.EntitySystems;
+using Content.Server.Power.Components;
+using Content.Shared.Chat;
+using Robust.Shared.Player;
+using Content.Server.PDA.Ringer;
+using Robust.Shared.Network;
+using Content.Server.DeviceNetwork;
+using Content.Server.DeviceNetwork.Components;
+using Content.Server.DeviceNetwork.Systems;
+using Content.Shared.DeviceNetwork;
+
+
+namespace Content.Server.CartridgeLoader.Cartridges;
+
+public sealed class MessagesCartridgeSystem : EntitySystem
+{
+    [Dependency] private readonly CartridgeLoaderSystem _cartridgeLoaderSystem = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly GameTicker _gameTicker = default!;
+    [Dependency] private readonly MessagesServerSystem _messagesServerSystem = default!;
+    [Dependency] private readonly DeviceNetworkSystem _deviceNetworkSystem = default!;
```

Comment:
You need to subscribe to the `CartridgeDeviceNetPacketEvent` instead of the `DeviceNetworkPacketEvent` as well as subscribe to the `CartridgeAddedEvent` and call  `_cartridgeLoaderSyste.RegisterBackgroundProgram`. 

This will cause the cartridge to use the PDAs device network connection instead of its own, and it will ensure that the cartridge is only connected to the network when it's installed or inserted inside a PDA.
You will need to use the `PDA` frequency for messaging which means you will have to check if it's a syndicate cartridge or not. I'd just use an int called faction or something. Calling it encryption key is confusing.

If the cartridge has its own device network connection it'll still be connected when not inserted into a PDA. You can listen to the `CartridgeActivatedEvent` and `CartridgeDeactivatedEvent` and connect and disconnect from the network there if you really want to have the cartridge use its own connection.

Reply:
I think I'll go cartridges having their own connections.
For one thing, I personally like it better to use a unique frequency, so unless it's for some reason needed otherwise, I'll stick with that.
For another, it's gonna make it a bit easier to potentially expand the network with non-pda messagers in the future.
Either way, thank you for the instructional feedback.